compiler-plugin/src/main/java/io/ballerina/stdlib/jwt/compiler/staticcodeanalyzer/JWTCipherAlgorithmAnalyzer.java

```diff
@@ -0,0 +1,190 @@
+    private static final String ALGORITHM = "algorithm";
+    private static final String NONE = "NONE";
+
+    public JWTCipherAlgorithmAnalyzer(Reporter reporter) {
+        this.reporter = reporter;
+    }
+
+    /**
+     * Analyzes jwt:issue() calls and reports usage of signatureConfig.algorithm == NONE.
+     * It checks inline mappings, function-local variables, and module-level variables.
+     *
+     * @param context the syntax node analysis context for reporting issues
+     */
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        FunctionCallExpressionNode functionCall = (FunctionCallExpressionNode) context.node();
+
+        // Check if the function is qualified with the module name
+        if (!(functionCall.functionName() instanceof QualifiedNameReferenceNode qualifiedName)) {
+            return;
+        }
+
+        // Verify the module and function name
+        if (!JWT.equals(qualifiedName.modulePrefix().text()) || !ISSUE.equals(qualifiedName.identifier().text())) {
+            return;
+        }
+
+        for (FunctionArgumentNode arg : functionCall.arguments()) {
+            if (!(arg instanceof PositionalArgumentNode posArg)) {
+                continue;
+            }
+            ExpressionNode expr = posArg.expression();
+            if (expr instanceof MappingConstructorExpressionNode mappingConstructor) {
+                if (isInsecureInlineMappingLiteral(mappingConstructor)) {
+                    reporter.reportIssue(
+                            getDocument(context.currentPackage().module(context.moduleId()), context.documentId()),
+                            context.node().location(),
+                            JWTRule.AVOID_WEAK_CIPHER_ALGORITHMS.getId()
+                    );
+                }
+            } else if (expr instanceof SimpleNameReferenceNode varRef) {
+                if (isInsecureVariableReference(varRef)) {
+                    reporter.reportIssue(
+                            getDocument(context.currentPackage().module(context.moduleId()), context.documentId()),
+                            context.node().location(),
+                            JWTRule.AVOID_WEAK_CIPHER_ALGORITHMS.getId()
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Determines if a variable reference refers to an IssuerConfig record with signatureConfig.algorithm == NONE.
+     * It walks up the AST to check function bodies and module-level declarations.
+     *
+     * @param varRef the simple name reference node pointing to the variable
+     * @return true if the variable's initializer mapping contains algorithm NONE, false otherwise
+     */
+    private boolean isInsecureVariableReference(SimpleNameReferenceNode varRef) {
+        String varName = varRef.name().text();
+        Node current = varRef.parent();
+        while (current != null) {
+            // Case 1: variable declared inside a function body
```

Comment:
This comment isn't really useful here, would make sense where this function is called.

This can end up being confusing, how many cases are there, what are the other cases, etc.